app/views/quizzes/new.html.erb

```diff
@@ -1,45 +1,102 @@
-<div class="container">
+<div class="container my-5">
+
   <% if @quiz.errors.any? %>
-    <div class="row text-bg-warning" id="error_explanation">
-      <h2><%= pluralize(@quiz.errors.count, "error") %> prevented this quiz from being saved:</h2>
+    <div class="row text-bg-warning mb-4" id="error_explanation">
+      <div class="col">
+        <h2 class="text-center">
+          <%= pluralize(@quiz.errors.count, "error") %> prevented this quiz from being saved:
+        </h2>
+      </div>
     </div>
     <% @quiz.errors.each do |error| %>
-      <div class="row text-bg-warning">
-        <div class="col-2"></div><div class="col-8"><%= error.full_message %></div>
+      <div class="row mb-2">
+        <div class="col-12">
+          <div class="alert alert-warning">
+            <%= error.full_message %>
+          </div>
+        </div>
       </div>
     <% end %>
   <% end %>
 
-  <div class="row">
-    <h1>Generate New Quiz</h1>
+  <div class="row mb-4">
+    <div class="col text-center">
+      <h1 class="display-4 text-primary">Generate New Quiz</h1>
+    </div>
   </div>
 
-  <%= form_with model: @quiz, method: :post do |f| %>
-    <div class="form-group">
-      <%= f.label :topic %>
-      <%= f.text_field :topic, class: 'form-control' %>
+  <%= form_with model: @quiz, method: :post, class: 'needs-validation' do |f| %>
```

Comment:
I think the issue with UI not rendering properly might be overlapping HTML elements and the erb code. I notice all the form elements are wrapped nicely in div, but the fact that the outer form with and it's corresponding <% end %> tag is not nested might be an issue. Will check this later when I clone your branch.